lib/lib-doi/src/test/java/org/sagebionetworks/doi/datacite/TranslatorTest.java

```diff
@@ -0,0 +1,190 @@
+						"            <nameIdentifier schemeURI=\"http://www.insi.org/\" nameIdentifierScheme=\"ISNI\">9876-5432-10123-456X</nameIdentifier></creator>"
+		)));
+		Element creator = (Element)dom.getElementsByTagName("creator").item(0);
+
+		DoiCreator expected = new DoiCreator();
+		expected.setCreatorName("Author");
+
+		DoiNameIdentifier id1 = new DoiNameIdentifier();
+		id1.setNameIdentifierScheme(NameIdentifierSchemes.ORCID);
+		id1.setIdentifier("0123-4567-8987-789X");
+		DoiNameIdentifier id2 = new DoiNameIdentifier();
+		id2.setNameIdentifierScheme(NameIdentifierSchemes.ISNI);
+		id2.setIdentifier("9876-5432-10123-456X");
+
+		List<DoiNameIdentifier> ids = new ArrayList<>();
+		ids.add(id1);
+		ids.add(id2);
+
+		expected.setNameIdentifiers(ids);
+		// Unit under test
+		assertEquals(expected, translator.getCreator(creator));
+	}
+
+	@Test
+	public void getCreatorsTest() throws Exception {
+		DataciteTranslator translator = new DataciteTranslator();
+
+		Document dom = documentBuilder.parse(new InputSource(new StringReader("<creators><creator><creatorName>Author 1</creatorName></creator><creator><creatorName>2, Author</creatorName></creator></creators>")));
+		List<DoiCreator> expected = new ArrayList<>();
+		DoiCreator creator1 = new DoiCreator();
+		creator1.setCreatorName("Author 1");
+		DoiCreator creator2 = new DoiCreator();
+		creator2.setCreatorName("2, Author");
+		expected.add(creator1);
+		expected.add(creator2);
+
+		assertEquals(expected, translator.getCreators(dom));
+	}
+
+	@Test
+	public void getTitlesTest() throws Exception {
+		DataciteTranslator translator = new DataciteTranslator();
+
+		Document dom = documentBuilder.parse(new InputSource(new StringReader("<titles><title>Title 1</title><title>Title 2</title></titles>")));
+		List<DoiTitle> expected = new ArrayList<>();
+		DoiTitle t1 = new DoiTitle();
+		DoiTitle t2 = new DoiTitle();
+		t1.setTitle("Title 1");
+		t2.setTitle("Title 2");
+		expected.add(t1);
+		expected.add(t2);
+
+		assertEquals(expected, translator.getTitles(dom));
+	}
+
+	@Test
+	public void getPubYearTest() throws Exception {
+		DataciteTranslator translator = new DataciteTranslator();
+		Document dom = documentBuilder.parse(new InputSource(new StringReader("<publicationYear>1929</publicationYear>")));
+		long expected = 1929;
+		assertEquals(expected, translator.getPublicationYear(dom));
+	}
+
+	@Test
+	public void getResourceTypeTest() throws Exception {
+		DataciteTranslator translator = new DataciteTranslator();
+		Document dom = documentBuilder.parse(new InputSource(new StringReader("<resourceType resourceTypeGeneral=\"Dataset\"></resourceType>")));
+
+		DoiResourceType expected = new DoiResourceType();
+		expected.setResourceTypeGeneral(DoiResourceTypeGeneral.Dataset);
+		assertEquals(expected, translator.getResourceType(dom));
+	}
+
+
+	// Tests the entire class
+	@Test
+	public void translateEntireDomTest() throws Exception{
```

Comment:
I'm not sure if this test belongs here, or if it would make more sense in an integration test class or elsewhere.